packages/compass-e2e-tests/helpers/test-runner-context.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  getConnectionTitle,
+  type ConnectionInfo,
+} from '@mongodb-js/connection-info';
+import path from 'path';
+import electronPath from 'electron';
+import electronPackageJson from 'electron/package.json';
+// @ts-expect-error no types for this package
+import { electronToChromium } from 'electron-to-chromium';
+import type { MongoClusterOptions } from 'mongodb-runner';
+
+if (typeof electronPath !== 'string') {
+  throw new Error(
+    'Running e2e tests in an unsupported runtime: `electronPath` is not a string'
+  );
+}
+
+// TODO: Probably time to use some arg parser for this already
+export const ALLOWED_RUNNER_ARGS = [
+  '--test-compass-web',
+  '--no-compile',
+  '--no-native-modules',
+  '--test-packaged-app',
+  '--disable-start-stop',
+  '--bail',
+];
+
+/**
+ * Variables used by a special use-case of running e2e tests against a
+ * cloud(-dev).mongodb.com URL. If you're changing anything related to these,
+ * make sure that the tests in mms are also updated to account for that
+ */
+export const TEST_ATLAS_CLOUD_EXTERNAL_URL =
+  process.env.TEST_ATLAS_CLOUD_EXTERNAL_URL;
+export const TEST_ATLAS_CLOUD_EXTERNAL_COOKIES_FILE =
+  process.env.TEST_ATLAS_CLOUD_EXTERNAL_COOKIES_FILE;
+export const TEST_ATLAS_CLOUD_EXTERNAL_GROUP_ID =
+  process.env.TEST_ATLAS_CLOUD_EXTERNAL_GROUP_ID;
+const TEST_ATLAS_CLOUD_EXTERNAL_DEFAULT_CONNECTIONS: ConnectionInfo[] | null =
+  JSON.parse(process.env.TEST_ATLAS_CLOUD_DEFAULT_CONNECTIONS ?? 'null');
```

Comment:
@syn-zhu FYI, I'm changing this to be a stringified json array with connection following ConnectionInfo type schema because we're refactoring these STRING_1 / NAME_1 variables to be always derived from a list of connection info objects